src/flamegraph/attrs.rs

```diff
@@ -83,44 +88,35 @@ impl FuncFrameAttrsMap {
 
 /// Attributes to set on the SVG elements of a frame
 #[derive(PartialEq, Eq, Debug, Default)]
-pub(super) struct FrameAttrs {
+pub struct FrameAttrs {
     /// The text to include in the `title` element.
     /// If set to None, the title is dynamically generated based on the function name.
     pub(super) title: Option<String>,
 
-    pub(super) g: GElementAttrs,
-    pub(super) a: AElementAttrs,
+    pub(super) attrs: IndexMap<String, String>,
 }
 
-/// Attributes to set on the SVG `g` element.
-/// Any of them set to `None` will get the default value.
-#[derive(PartialEq, Eq, Debug, Default)]
-pub(super) struct GElementAttrs {
-    /// Will not be included if None
-    pub(super) class: Option<String>,
-
-    /// Will not be included if None
-    pub(super) id: Option<String>,
-
-    /// Extra attributes to include
-    pub(super) extra: Vec<(String, String)>,
-}
-
-/// Attributes to set on the SVG `a` element
-#[derive(PartialEq, Eq, Debug, Default)]
-pub(super) struct AElementAttrs {
-    /// If set to None the `a` tag will not be added
-    pub(super) href: Option<String>,
-
-    /// Defaults to "_top"
-    pub(super) target: Option<String>,
-
-    /// Extra attributes to include
-    pub(super) extra: Vec<(String, String)>,
-}
+impl FrameAttrs {
+    fn add_attr(&mut self, func: &str, name: String, value: String) {
+        match self.attrs.entry(name) {
+            Entry::Occupied(mut e) => {
+                warn!(
+                    "duplicate attribute `{}` in nameattr file for `{}`; replacing value \"{}\" with \"{}\"",
```

Comment:
So we'll get this warning if any of the standard attributes are overwritten then I take it?

Reply:
The container elements for a function won't have any attributes except those that are added using the nameattr file. This will warn if any of those attributes are duplicated.

Reply:
Ah, I think I misread the code a little. I thought this map was going to hold _all_ the attributes, but not realize that there is only really one default attribute now, and that is `target`. I don't know where you're planning on adding that, but let's try to make sure we don't emit this warning when the user tries to set `target`.